### PR TITLE
Helper - Modify ErrorSchema.json

### PR DESCRIPTION
**Pull request by leandro0day**
We added a description to each of the properties of the ErrorResponse.json  and "additionalProperties: false".
-add ErrorSchema response type to all endpoint.

### CODE REVIEW

File: src/routes/brands/updateBrand.ts [hunk 1]
```diff
@@ -25,14 +27,13 @@ const handler: RouteHandler<UpdateBrandRoute> = async (req, reply) => {
     const brandId = req.params.id;
```
Comment (AlvisonHunterArnuero) [inline src/routes/brands/updateBrand.ts]: **suggestion:** Could you please apply destructuring to the **id** with an alias...
```suggestion
const {id:brandId} = req.params;
```

File: src/routes/brands/updateBrand.ts [hunk 2]
```diff
@@ -25,14 +27,13 @@ const handler: RouteHandler<UpdateBrandRoute> = async (req, reply) => {
...
     const newBrandName = req.body.name;
```
Comment (AlvisonHunterArnuero) [inline src/routes/brands/updateBrand.ts]: **suggestion:** Could you please apply destructuring to the **name** with an alias...
```suggestion
const {name:newBrandName} = req.body;
```

File: src/routes/areas/deleteArea.ts [hunk 3]
```diff
@@ -3,30 +3,34 @@ import {
...
   Body: Area;
   Params: AreaRequestParams;
   Reply: Reply;
 };
 
-const url = '/area/:id';
+const url = "/area/:id";
 
 const handler: RouteHandler<DeleteAreaRoute> = async (req, reply) => {
   const areaId = req.params.id;
```
Comment (AlvisonHunterArnuero) [inline src/routes/areas/deleteArea.ts]: **suggestion:** Could you please apply destructuring to the **id** with an alias...
```suggestion
const {id:areaId} = req.params;
```

**Review by AlvisonHunterArnuero — CHANGES_REQUESTED** [review]
@leandro0day Please take care of the comments if suggestions are feasible for Fastify standards.

File: src/routes/areas/createArea.ts [hunk 4]
```diff
@@ -38,33 +40,24 @@ export const handler: RouteHandler<CreateAreaRoute> = async (req, reply) => {
     reply.status(200).send();
   } catch (error) {
     return reply.status(500).send({
-      error: {
-        code: 'unknown',
-        message: `An unknown error occurred when trying to create an Area. Error: ${error}`
-      }
+      error: `${error}`,
```
Comment (Ralexs0096) [inline src/routes/areas/createArea.ts]: ummm I thought we were returning an error with this structure:

```js
{
error: ErrorSchema
}
```

am I wrong?  🤔

File: src/routes/areas/deleteArea.ts (hunk 3 above)
Comment (Ralexs0096) [reply src/routes/areas/deleteArea.ts]: it's a [nitpick](https://conventionalcomments.org/) @leandro0day (non-blocking by nature)

File: src/routes/areas/updateArea.ts [hunk 5]
```diff
@@ -53,15 +57,7 @@ const schema = {
...
-    400: {
-      title: 'InvalidArea',
-      description: 'Invalid or missing Area data.',
-      type: 'object',
-      required: ['name'],
-      properties: {
-        error: {}
-      }
-    }
+    400: ErrorSchemaJson
```
Comment (Ralexs0096) [inline src/routes/areas/updateArea.ts:60]: Nice! it's more clear now, right? 😄

File: src/routes/brands/fetchAllBrands.ts [hunk 6]
```diff
@@ -53,46 +49,8 @@ export const schema = {
...
-              type: "string",
-            },
-            message: {
-              type: "string",
-            },
-          },
-        },
-      },
-    },
+    204: {},
```
Comment (Ralexs0096) [inline src/routes/brands/fetchAllBrands.ts]: ❗ why are we changing this status code? 🤔

File: src/routes/brands/updateBrand.ts [hunk 7]
```diff
@@ -70,46 +72,8 @@ const schema = {
...
-              type: "string",
-            },
-            message: {
-              type: "string",
-            },
-          },
-        },
-      },
-    },
+    404: ErrorSchemaJson,
```
Comment (Ralexs0096) [inline src/routes/brands/updateBrand.ts]: in this case, we don't want to change the entire object, Only the error at line 79.

File: src/routes/areas/createArea.ts [hunk 8]
```diff
@@ -40,9 +40,11 @@ export const handler: RouteHandler<CreateAreaRoute> = async (req, reply) => {
     reply.status(200).send();
   } catch (error) {
     return reply.status(500).send({
-      error: `${error}`,
-      statusCode: 500,
-      message: "An unknown error occurred when trying to create an Area."
+      error: {
+        error: `${error}`,
+        code: "500",
```
Comment (Ralexs0096) [inline src/routes/areas/createArea.ts]: the code is not the statusCode, it's something like InvalidId, NotAllowed, etc.
in this case, we want a code like: **unknown**

File: src/routes/areas/updateArea.ts [hunk 9]
```diff
@@ -3,36 +3,38 @@ import {
...
   const areaResponse = await AreaModel.query().findById(idArea);
 
   if (!areaResponse) {
     return reply.status(404).send({
-      error: "Not found",
-      statusCode: 404,
-      message: 'This area does not exist'
+      error: {
+        error: "Not found",
+        code: "",
```
Comment (Ralexs0096) [inline src/routes/areas/updateArea.ts]: what about this code?
probably: **NotFound**

File: src/routes/brands/createBrand.ts [hunk 10]
```diff
@@ -39,9 +39,11 @@ export const handler: RouteHandler<CreateBrandRoute> = async (req, reply) => {
     reply.status(201).send(createdBrands);
   } catch (error) {
     return reply.status(500).send({
-      error: `${error}`,
-      statusCode: 500,
-      message: "An unknown error occurred when trying to create an Brand.",
+      error: {
+        error: `${error}`,
+        code: "unknown",
```
Comment (Ralexs0096) [inline src/routes/brands/createBrand.ts]: Correct! 👍🏻

File: src/routes/brands/createBrand.ts [hunk 11]
```diff
@@ -39,9 +39,11 @@ export const handler: RouteHandler<CreateBrandRoute> = async (req, reply) => {
...
+        message: "An unknown error occurred when trying to create an Brand.",
```
Comment (Ralexs0096) [inline src/routes/brands/createBrand.ts]: 👀 : a Brand.

File: src/routes/brands/createBrand.ts [hunk 12]
```diff
@@ -58,7 +60,15 @@ export const schema = {
       type: "array",
       description: "Brand(s) successfully created",
     },
-    500: ErrorSchemaJson,
+    500: {
+      title: "Error",
+      description: "An unknown error occurred when trying to create an Brand.",
```
Comment (Ralexs0096) [inline src/routes/brands/createBrand.ts]: same here

File: src/routes/brands/createBrand.ts (hunk 10 above)
Comment (Ralexs0096) [reply src/routes/brands/createBrand.ts]: please capitalize

File: src/routes/brands/deleteBrand.ts [hunk 13]
```diff
@@ -23,14 +23,16 @@ const url = "/brand/:id";
...
     const brandToDelete = await BrandModel.query().findById(brandId);
 
     if (!brandToDelete) {
       return reply.status(404).send({
-        error: "Not fount",
-        statusCode: 404,
-        message: "This brand does not exist",
+        error: {
+          error: "Not fount",
+          code: "notFount",
```
Comment (Ralexs0096) [inline src/routes/brands/deleteBrand.ts]: please capitalize it

File: src/routes/areas/deleteArea.ts [hunk 14]
```diff
@@ -3,56 +3,62 @@ import {
...
-  const areaId = req.params.id;
+  const { id: areaId } = req.params;
   const areaResponse = await AreaModel.query().findById(areaId);
 
   if (!areaResponse) {
     return reply.status(404).send({
-      message: 'This area does not exist'
+      error: {
+        error: "Not Found",
+        code: "notFound",
```
Comment (Ralexs0096) [inline src/routes/areas/deleteArea.ts]: Capitalize it too

File: src/routes/areas/fetchAllAreas.ts [hunk 15]
```diff
@@ -3,40 +3,41 @@ import {
...
   Reply: Reply;
 };
 
-const url = '/area';
+const url = "/area";
 
 export const handler: RouteHandler<FetchAllAreas> = async (req, reply) => {
-  const areas = await AreaModel.query().select('name');
-  reply.status(201).send(areas);
+  const areas = await AreaModel.query().select("name");
```
Comment (Ralexs0096) [inline src/routes/areas/fetchAllAreas.ts]: add a try-catch and return an 500 unknown with our error Object

File: src/routes/areas/fetchAllAreas.ts [hunk 16]
```diff
@@ -3,40 +3,41 @@ import {
...
 };
 
 export const schema = {
-  operationId: 'fetchAllAreas',
-  tags: ['Area'],
-  summary: 'Fetch All Areas',
+  operationId: "fetchAllAreas",
+  tags: ["Area"],
+  summary: "Fetch All Areas",
   response: {
```
Comment (Ralexs0096) [inline src/routes/areas/fetchAllAreas.ts:39]: this will need a 500 status when you add the try-catch

File: src/routes/areas/updateArea.ts [hunk 17]
```diff
@@ -3,32 +3,38 @@ import {
...
+  const { name: newAreaName } = req.body;
 
   const areaResponse = await AreaModel.query().findById(idArea);
 
   if (!areaResponse) {
     return reply.status(404).send({
-      message: 'This area does not exist'
+      error: {
+        error: "Not Found",
+        code: "notFound",
```
Comment (Ralexs0096) [inline src/routes/areas/updateArea.ts]: capitalize it, please

File: src/routes/areas/updateArea.ts [hunk 18]
```diff
@@ -3,32 +3,38 @@ import {
...
 type UpdateAreaRoute = {
   Body: Area;
   Params: AreaRequestParams;
   Reply: Reply;
 };
 
-const url = '/area/:id';
+const url = "/area/:id";
 
 const handler: RouteHandler<UpdateAreaRoute> = async (req, reply) => {
```
Comment (Ralexs0096) [inline src/routes/areas/updateArea.ts:25]: need try-catch

File: src/routes/brands/deleteBrand.ts [hunk 19]
```diff
@@ -39,8 +42,9 @@ const handler: RouteHandler<DeleteBrandRoute> = async (req, reply) => {
   } catch (error) {
     return reply.status(500).send({
       error: {
-        code: "unknown",
-        message: `An unknown error occurred when trying to delete a brand. Error: ${error}`,
+        error: `${error}`,
+        code: "Unknown",
```
Comment (Ralexs0096) [inline src/routes/brands/deleteBrand.ts:46]: consider this an standard please 😃 , always capitalized

File: src/routes/brands/updateBrand.ts [hunk 20]
```diff
@@ -22,15 +24,16 @@ const url = "/brand/:id";
...
 
-     const brandToUpdate = await BrandModel.query().findById(brandId);
+    const brandToUpdate = await BrandModel.query().findById(brandId);
 
     if (!brandToUpdate) {
       return reply.status(404).send({
         error: {
-          code: "unknow",
+          error: "Not Found",
+          code: "notFound",
```
Comment (Ralexs0096) [inline src/routes/brands/updateBrand.ts]: same here

File: src/routes/brands/deleteBrand.ts (hunk 13 above)
Comment (Ralexs0096) [reply src/routes/brands/deleteBrand.ts]: I just wanted to let you know that this suggestion was not applied. @leandro0day, please make sure always to use all the suggestions.

**Pull request merged**